mllib/src/main/scala/org/apache/spark/mllib/feature/HashingTF.scala

```diff
@@ -47,7 +47,7 @@ class HashingTF(val numFeatures: Int) extends Serializable {
   /**
    * Transforms the input document into a sparse term frequency vector.
    */
-  def transform(document: Iterable[_]): Vector = {
+  def transform(document: Iterable[_]): SparseVector = {
```

Comment:
A `SparseVector` is a `Vector`. This promises the more specific subclass, but is it necessary?